### DISCUSSION TITLE
如何把钱包里的所有sui转出，使用官方的js

### DISCUSSION THREAD

**Discussion by lly835** [RandyPen/sui-move-intro-course-zh#3] [Q&A] [answered]
https://github.com/MystenLabs/sui/tree/main/sdk/typescript
```
const tx = new TransactionBlock();
const [coin] = tx.splitCoins(tx.gas, [tx.pure(1000)]);
```
官方的例子，要如何设置gas才能将钱包里sui余额全部转出。
或者，能不能设置gas为另一个钱包代付?

**Comment by RandyPen** [comment, marked as answer]
1. gas是动态调整的，dryrun也仅是用于预估gas费用。 https://docs.sui.io/build/sui-gas-charges
2. 可以设置代支付的 https://docs.sui.io/build/prog-trans-ts-sdk#sponsored-transaction-blocks

**Reply by lly835** [reply]
如果设置代付，signAndExecuteTransactionBlock()的参数  要怎么写呢，2个signer?

**Reply by RandyPen** [reply]
亲，你来到了没有知识的荒漠。
1. 可以自己查看文档和源代码，探索解决；
2. 小众场景我也还没遇到，不在教程里。如果确实有需要，可以付费200U提供技术咨询，我也需要花时间研究。
